Source/Cogworks.Umbraco.Essentials/Models/ImageBreakpoints.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Cogworks.Umbraco.Essentials.Models
+{
+    public class ImageBreakpoints
```

Comment:
What about creating dedicated class that simplify usage:

1) https://dotnetfiddle.net/3mG4Bd
2) https://dotnetfiddle.net/IZbEjm

Reply:
I like this idea! Both solutions will work, but here we'll have less coupling and dependencies when we'll be dealing directly on the object/class. Definitely worth considering.

Reply:
I did try this today and I honestly I don't know/see how this is can be simpler/better than a dumb class as the one that is in place, could any of you jump in a call with me and show me your idea. I can merge this as I got the 2 green ticks but I'm curious. please let me know asap as I do want to merge this PR this week. 👍

Reply:
Lets wait for tomorrow

Reply:
Its just for how to work with ImageBreakpoints, this class will behaves like object with configurations (we will have options - depends on approach we take - to pass list/dictionary of breakpoints or add it via method). From user perspective will be easily to know how to work with it.

For example:

```csharp
var imageBreakpoints = new ImageBreakpoints(new Dictionary<string, string>() { ["key"] = "value" });

imageBreakpoints.AddBreakpoint("key2", "value2"); //depends if we want to give abilities for extending

foreach(var imageBreakpoint in imageBreakpoints.GetBreakpoints())
{
    // logic
}

// or

foreach(var imageBreakpoint in imageBreakpoints)
{
    // logic
} 

if (imageBreakpoints.HasAny())
{
  // logic
}
```

Personal I would go with immutable approach for assigning object with list/dictionary without ability to add/remove new entries.

The constructor of this object will ensure on user to pass list/dictionary of configuration.

Reply:
Sorry this method `GetBreakpoints() should be:

```csharp
public IEnumerable<KeyValuePair<string, string>> GetBreakpoints()
        => _breakpoints.ToArray();
```

Reply:
Full example: https://dotnetfiddle.net/Dzca6E

Reply:
Another approach could be domain crops, example here: https://dotnetfiddle.net/qkscUW but this one is not generic for all.

Reply:
I will create separate PR for the last domain example usage.